src/common/misc.c

```diff
@@ -681,6 +683,46 @@ daos_prop_free(daos_prop_t *prop)
 	D_FREE_PTR(prop);
 }
 
+static bool
+daos_prop_str_valid(d_string_t str, const char *prop_name, size_t max_len)
+{
+	size_t len;
+
+	if (str == NULL) {
+		D_ERROR("invalid NULL %s\n", prop_name);
+		return false;
+	}
+	len = strlen(str);
```

Comment:
Just noticed this. Are we sure that str will always be null terminated? If not we should use a strnlen here (really we should probably use it regardless).

Reply:
I've seen codacy complain about strnlen as well but yeah, it's good practice, particularly if there is a known maximum length to use strnlen.

Reply:
Everywhere daos_prop_valid is used on an internally-generated string, we know the string is null terminated. But it's used on user inputs as well. I'll change it to the safer strnlen.